apollo-router/src/context.rs

```diff
@@ -38,6 +39,9 @@ pub struct Context {
     #[serde(skip)]
     #[serde(default = "Instant::now")]
     pub(crate) created_at: Instant,
+
+    #[serde(skip)]
+    busy_timer: Arc<Mutex<BusyTimer>>,
```

Comment:
Is the `Mutex` avoidable? Atomics...

Reply:
I wanted to go with atomics first, but having to set the `start` field complicates it. At least this could be simplified by using a parking lot mutex instead of a futures aware mutex

Reply:
as a side note, I'd like to review our uses of mutexes, and see which ones could move away from futures aware locks,  it might be faster [as outlined here](https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use). I pushed hard for removing a lot of them and moving towards futures mutex in the past because we had locks all over the place, but now that the code is much more stable, this is something we can revisit